lib/packwerk/package_set.rb

```diff
@@ -28,8 +28,12 @@ def load_all_from(root_path, package_pathspec: nil)
       def package_paths(root_path, package_pathspec)
```

Comment:
I'm a bit concerned about the number of responsibilities the PackageSet class has. I think think the functionality of the `PackageSet .load_all_from(root_path, package_pathspec: nil)` class method  and by extension the `PackageSet .package_paths(root_path, package_pathspec)` class method should be extracted into a separate class. IMO the functionality should be ideally in a separate class that is responsible for loading the packages from the YAML. We can then call that class from `PackageSet .load_all_from(root_path, package_pathspec: nil)`.

I'm bringing this up because you are changing the Package loading behavior, [so it makes sense to leave it in a better state than you found it](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html).

Reply:
I'm expanding the package loading behaviour by supporting arrays.  I'm not changing behaviour for existing inputs nor changing the reponsabilies of PackageSet.  Refactoring PackageSet is not going to make my change easier or clearer.  It's not going to affect how other bits of code call PackageSet, either.

Refactoring the class as part of this PR will distract from my small change; I certainly don't want to set this work aside, as it is already a precusory PR for a larger set of changes.

Reply:
> I'm expanding the package loading behavior by supporting arrays.

IMO this constitutes different behavior because it changes how we interact with this class. Furthermore, adding the new `glob_patterns` operation makes the code more complex, not less.

Making classes more complex just a little bit will add up over time and make the code harder to maintain. Since you are adding complexity, you are not leaving this class stagnant nor better. I understand that it might seem silly to take such a hard stance for a minuscule addition to the class, but I think it is important to take these small changes seriously because they add up over time.

> Refactoring PackageSet is not going to make my change easier or clearer. 
> It's not going to affect how other bits of code call PackageSet, either.
> Refactoring PackageSet is not going to make my change easier or clearer. It's not going to affect how other bits of code call PackageSet, either.

It is important to separate classes by making the code easier to understand; otherwise, we could put all code into one class. I already have a hard time understanding what the PackageSet class is intended for, and the proposed change will not make that easier. 

If you want to address this in a separate PR, I'm ok with that because I agree with your assertion that this change would distract from the actual PR, and I don't want to block you on your larger work.

Reply:
I have thought about your arguments and I understand your point of view.  I don't think we're at a point where this incremental, backwards-compatible change warrants a refactor of PackageSet.  The class remains an enumerator for packages that also has a factory-like entrypoint.  I'm not against splitting it up at a later point if the requirements change enough.